base_location/models/res_partner.py

```diff
@@ -34,6 +34,17 @@ def _onchange_country_id(self):
         res = super()._onchange_country_id()
         if self.zip_id and self.zip_id.city_id.country_id != self.country_id:
             self.zip_id = False
+        if self.country_id:
+            city_zip_domain = {
+                "zip_id": [("city_id.country_id", "=", self.country_id.id)]
+            }
+            if isinstance(res, dict):
+                if "domain" in res and isinstance(res["domain"], dict):
```

Comment:
Maybe better to use `zip_domain = res.setdefault('domain', {}`?

Reply:
ok Thnak you,

I applied your suggestion